src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs

```diff
@@ -350,11 +350,13 @@ public override BoundNode VisitGotoStatement(BoundGotoStatement node)
                 }
                 else if (sourceStart > usingStart && targetStart < usingStart)
                 {
-                    // Backwards jump, so we must have already seen the label
-                    Debug.Assert(_labelsDefined.ContainsKey(node.Label));
+                    // Backwards jump, so we must have already seen the label, or it must be a switch case label. If it is a switch case label, we know
+                    // that either the user received an error for having a using declaration at the top level in a switch statement, or the label is a valid
+                    // target to branch to.
+                    Debug.Assert(_labelsDefined.ContainsKey(node.Label) || node.Label is SourceLabelSymbol { SwitchCaseLabelConstant: not null });
```

Comment:
>node.Label is SourceLabelSymbol { SwitchCaseLabelConstant: not null } [](http://example.com/codeflow?start=75&length=69)

Is this robust enough test? Will it handle ```default``` label? Consider adding switch labels to the map when we visit switch sections instead. The value in the map could be either a block. or a switch statement.

Reply:
It will handle the `default` label, I will add a test. I originally did this suggestion, but I thought it was a bit of overkill. If you prefer, I can switch back to this approach though.

Reply:
Actually, I guess I was wrong. I will take this approach.